### ISSUE TITLE
New World - Failed to create buffer view and crashing

### ISSUE THREAD

**Issue by OfficialEsco** [doitsujin/dxvk#3698]
So ever since the New World Call of the Wilds DLC update i've been having some nasty consistent crashes at around the 1 hour mark. The most noticeable crash is the game picture stops updating and then i get disconnected from the game, however the game does not crash even after waiting 10min.
I've tried to look at the Proton Logs but i cant find anything among the millions of lines...
I did however grep for `err:`, `warn:` and `info:` lines and found these DXVK errors at the bottom of the log
It always starts with
```
D3D11: Cannot create texture
```
Then these
```
DxvkBufferView: Failed to create buffer view
```
Today i got a different error message but in the same ball park
```
DxvkImageView: Failed to create image view
```

So i wonder, is this a error message for the DXVK devs or the game devs that a texture is broken?

This is what i've tried:
- I've tested both x11 and Wayland
- I've tested both KDE and Gnome
- I've tested Proton Experimental, GE-Proton 8-21, Proton 8.0-4 and Proton 7.0-6
- I've tested DX11 and DX12
- I've tested a bunch of launch options
- I've reset the ShaderCache in steamapps/shadercache/1063730/DXVK_state_cache/
- I've checked that i'm not running out of VRAM
- I've deleted `SteamLibrary/steamapps/compatdata/1063730/pfx/drive_c/users/steamuser/AppData/*/AGS` as someone suggested for crashes
- I've "Deleted all data and applications inside this wineprefix" via ProtonTricks
- I've installed d3dcompiler_43 via ProtonTricks
- I've installed d3dcompiler_47 via ProtonTricks (Fixes fixme:d3dcompiler:skip_u32_unknown in the logs)
- I've installed dxvk via ProtonTricks
- I've reinstalled New World from scratch
- I've removed Mangohud and Gamemode from launch options
- PROTON_USE_WINED3D=1 makes it "The computer fails to meet one or more of the minimum specifications in order to run" because its 1280 MB VRAM by default. Workaround to set VideoMemorySize to 11GB in Registry, however the rendering is messed up and not worth it.
- I've tested vkd3d-proton via Protontricks
- I've tested nvidia-open
- I've tested PROTON_HIDE_NVIDIA_GPU=1
- I've tested ENABLE_PROTON_NVAPI=1

The ONE thing that DID work is to play differently in one old area, the log file was calm and i did not crash for 2.5 hours. Which probably indicates a issue with a texture?

### Software information
New World on any settings.

### System information
<details>
<summary>Click for System information</summary>
<br>

- Operating System: Arch Linux 
- KDE Plasma Version: 5.27.9
- KDE Frameworks Version: 5.111.0
- Qt Version: 5.15.11
- Kernel Version: 6.5.8-arch1-1 (64-bit)
- Graphics Platform: X11
- Processors: 32 × AMD Ryzen 9 5950X 16-Core Processor
- Memory: 31.3 GiB of RAM
- Graphics Processor: NVIDIA GeForce RTX 3080 Ti/PCIe/SSE2
- Manufacturer: ASUS

</details>

- GPU: Nvidia 3080 ti
- Driver: Proprietary and Open-Source kernel 535.113.01
- Wine version: Proton 7.0-6, 8.0-4 and GE-Proton 8-20
- DXVK version: No clue

### Apitrace file(s)
- Put a link here

For instructions on how to use apitrace, see: https://github.com/doitsujin/dxvk/wiki/Using-Apitrace

### Log files
The logs are TOO DAMN BIG (50mill lines and 5GB in 1 hour.
Most of the log is flodded with these
```
9644.436:0134:0374:trace:unwind:dump_unwind_info     0xa: pushq %rdi
```
[steam-1063730-err.log](https://github.com/doitsujin/dxvk/files/13182654/steam-1063730-err.log)
[steam-1063730-info.log](https://github.com/doitsujin/dxvk/files/13182655/steam-1063730-info.log)
[steam-1063730-err2.log](https://github.com/doitsujin/dxvk/files/13182669/steam-1063730-err2.log)

I currently have a 1.2GiB crash log file i can grep if anyone have something i can search for

**Comment by PatrickvL** [comment]
Could you elaborate on how "to play differently in one old area"? It seems rather crucial so a detailed description might give a hint to troubleshoot with

**Comment by OfficialEsco** [comment]
So to be specific, one day i decided to farm trees in Everfall while testing a change and waiting for the crash, in those 2 hours the highest place i went to was Myrkgard. 
The places i constantly crash in seems to be in elite zones in Brimstone and Elysium (new zone) and newer dungeons (might be Corrupted related)

**Comment by Blisto91** [comment]
> I've tested DX11 and DX12

Does the game support dx12?

> I currently have a 1.2GiB crash log file i can grep if anyone have something i can search for

The log should compress very well.
Alternatively you can run with `PROTON_LOG=-unwind`

**Comment by OfficialEsco** [comment]
Oh it did compress very well (55mb), but GitHub does not like anything over 25MB.
I also see i've removed the "dump_unwind_info" lines with this command
```
sed '/.*trace:unwind:dump_unwind_info.*/d' ./steam-1063730.log > ./steam-1063730_crash.log
```
so here is a cute 214 MiB unedited log where i got the new "Failed to create image view" crash
[steam-1063730.tar.gz](https://github.com/doitsujin/dxvk/files/13189960/steam-1063730.tar.gz)

New World does kind of support -dx12, but also not really as its been put on ice.

What does -unwind do? (guess i'll find out live)

**Comment by Blisto91** [comment]
> What does -unwind do? (guess i'll find out live)

It just makes it so the unwinds aren't logged.

> New World does kind of support -dx12, but also not really as its been put on ice.

Did you verify it was using d3d12 via vkd3d-proton then? If it crashes the same way.

**Comment by OfficialEsco** [comment]
> It just makes it so the unwinds aren't logged.

Ahaaaa ok i'll use this from now on

> Did you verify it was using d3d12 via vkd3d-proton then? If it crashes the same way.

How can i see if it uses DX12?
Mangohud still says DXVK, and i do notice a difference in performance

Update adding info from DX11 proton log:
```
info:  DXVK: v2.3-11-ge00db245574c7a5
info:  D3D11InternalCreateDevice: Maximum supported feature level: D3D_FEATURE_LEVEL_12_1
info:  D3D11InternalCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_1
```

Starting the game with -dx12
```
info:  DXVK: v1.10.3
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_1
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_1
```

> Edit: Do you know if there is a good area not too far in to the game that is decent to reproduce it or have you mostly tried the later areas. I'm not too familiar with the game.

Yeeeah thats kind of a issue, my daily run consists of doing Heliopolis (Brimstone) -> Westwall Nebet-Het (Brimstone) -> Castrum Principium (Brimstone), i will usually crash in Castrum, depending on other players i do Myrkgard in Shattered Mountains, Atalus Foundry in Ebonscale and Imperial Palace in Ebonscale. I also loot the chests in Isle of Zurvan and Dreaded Landing (i've crashed at the end of the cave here) in Elysian Wilds. All of these places are 65+ Areas (one which requires the DLC...)

I've also crashed in "The Depths" dungeon which i ran without issues before the DLC update.
Unsure if i've tested any other dungeons as i've jumped over to my VFIO VM to do stuff which requires not crashing.. (I've done some dungeons but crashed after getting out of the dungeon)
I'll take note of where exactly i'm crashing from now on as i thought this was a global map issue.

**Comment by Blisto91** [comment]
> Mangohud still says DXVK, and i do notice a difference in performance 

Then it's still d3d11 api. Feature level is a different thing. (I'm not technical enough to give a in depth explanation :grin: )


I will try to poke at the game a bit and see if i hit anything thanks :eyes:

**Comment by OfficialEsco** [comment]
Tested `PROTON_HIDE_NVIDIA_GPU=1` which gave horrible performance
[steam-1063730_crash.tar.gz](https://github.com/doitsujin/dxvk/files/13191660/steam-1063730_crash.tar.gz)
Crashed at the start of Dreaded Landing after doing Heliopolis -> Westwall -> Castrum
Weirdly DXVK still says version 1.10.3

**Comment by Blisto91** [comment]
1.10.3 is the version used with Proton 7.0-6 (actually it has more commits than released 1.10.3 but the version number will say that)

**Comment by OfficialEsco** [comment]
Oh my bad, forgot i was going to double check proton 7... okok so DXVK is bundled with Proton, so when i use GE-Proton 8-21 i'm on the latest version (according to patch notes)

**Comment by Blisto91** [comment]
I've hit the `D3D11: Cannot create texture:` errors a few times while leveling in the beginning of the game. Mostly when fast traveling to a settlement i think. But nothing fatal as i haven't encountered a crash yet.
The settlements seem to be quite heavy in general using a large mount of vram, at least on very high. And the allocated amount can stay high when traveling from place to place.

<details>
  <summary>screenshot</summary>

![Screenshot_20231028_002551](https://github.com/doitsujin/dxvk/assets/47954800/404c3b2e-07cd-4c8a-932b-6a6e003e6327)
</details>

**Comment by OfficialEsco** [comment]
Yeah i'm kinda unsure if its starting to be related, because from my 5 hour session log in Windsward/Everfall it does appear a few times with a lot of space between without being the cause of crash...

I've seen people suggest upping the swap file size for the settlements, but noticed no difference myself.

Also what a dedication starting from scratch in a game to troubleshoot a potential issue :p

I just checked journalctl for errors related to New World and i cannot see anything related to New World/NewWorld at 13:00 blank which is apparently when i crashed..
It does spam this tho while playing
```
Oct 28 12:59:06 esco pipewire-pulse[2416]: mod.protocol-pulse: 0x55924e63c0a0: [New World] overrun recover read:3644416 avail:80896 max:7680 skip:78976
```

**Comment by OfficialEsco** [comment]
OKAY SO, i've finally found something in the journalctl that matches the crash time frame. So DXVK might not be related, even tho those errors should probably be explained

<details>
<summary>Process 5642 (Main) of user 1000 dumped core.</summary>
<br>

```
Oct 28 20:43:05 esco systemd-coredump[17416]: [🡕] Process 5642 (Main) of user 1000 dumped core.
                                              
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/bin/wine64 without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/lsteamclient.dll.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/lsteamclient.dll.so
                                              Module /run/host/usr/lib/libnvidia-ml.so.535.113.01 without build-id.
                                              Module /run/host/usr/lib/libnvidia-ml.so.535.113.01
                                              Module /home/esco/.local/share/Steam/steamapps/common/Proton EasyAntiCheat Runtime/v2/lib64/easyanticheat_x64.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/kerberos.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/secur32.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winspool.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/libopenxr_loader.so.1.0.27 without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/libopenxr_loader.so.1.0.27
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/wineopenxr.dll.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/wineopenxr.dll.so
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winevulkan.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winevulkan.so
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winepulse.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winepulse.so
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winex11.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/winex11.so
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/ws2_32.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/win32u.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/win32u.so
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/bcrypt.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/libwine.so.1.0 without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/libwine.so.1.0
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/ntdll.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/ntdll.so
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/dnsapi.so without build-id.
                                              Module /home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/crypt32.so without build-id.
                                              Stack trace of thread 6034:
                                              #0  0x00007fef1900b3c1 n/a (/home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/ntdll.so + 0x513c1)
                                              #1  0x00007fef1900c2c5 n/a (/home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/ntdll.so + 0x522c5)
                                              #2  0x00007fef1a85c710 n/a (/run/host/usr/lib/libc.so.6 + 0x3e710)
                                              #3  0x00007fef1900c044 n/a (/home/esco/.local/share/Steam/compatibilitytools.d/GE-Proton8-21/files/lib64/wine/x86_64-unix/ntdll.so + 0x52044)
                                              #4  0x00007fef1a85c710 n/a (/run/host/usr/lib/libc.so.6 + 0x3e710)
                                              #5  0x00007fec680f9770 n/a (n/a + 0x0)
                                              ELF object binary architecture: AMD x86-64
```
</details>

**Comment by OfficialEsco** [comment]
Okay so, yeah i think the specific screen freeze crash is related to DXVK and the error causing the crash is `err:   DxvkBufferView: Failed to create buffer view:`
[steam-1063730.tar.gz](https://github.com/doitsujin/dxvk/files/13198833/steam-1063730.tar.gz)

Nothing in Journalctl like the Proton crash

**Comment by Blisto91** [comment]
These crashes are also happening in the regular open world right? Not only in raids or dungeons or what they are called. Just so I am sure.

**Comment by OfficialEsco** [comment]
The raids are in the open world so yes, but i've not crashed while questing, so i think its specific to the raid area, i actually crashed twice in a very short time yesterday in Dreaded Landing/Tribunal without a raid party

**Comment by Blisto91** [comment]
Lovely :grimacing:

**Comment by Blisto91** [comment]
I've gotten to the Elysian Wilds and peeked around a bit in Dreaded Landing/Tribunal (and fertilizing the ground with my corpses) but haven't had a crash yet. But will try some more

**Comment by OfficialEsco** [comment]
OK, thank you for the testing, i think i've fixed the problem by throwing money at it, my new AMD 7900XTX have not crashed doing these activities...
TLDR for anyone getting these weird crashes in New World, throw your expensive Nvidia GPU out the window and go with team Red...
